### ISSUE TITLE
Data Explorer Code Completion Not working

### ISSUE THREAD

**Issue by devtrix** [rethinkdb/rethinkdb#7109]
I have version 2.4.3 for windows and the code completion in the Data Explorer doesnt work.  When looking in the browser console I see there is JQuery error.
jquery-3.6.1.min.js?v=2.4.3:2 jQuery.Deferred exception: Cannot read properties of undefined (reading 'webkit') TypeError: Cannot read properties of undefined (reading 'webkit')
    at Typeahead.listen (http://127.0.0.1:8080/js/bootstrap/bootstrap-typeahead.js?v=2.4.3:202:21)
    at new Typeahead (http://127.0.0.1:8080/js/bootstrap/bootstrap-typeahead.js?v=2.4.3:40:10)
    at HTMLInputElement.<anonymous> (http://127.0.0.1:8080/js/bootstrap/bootstrap-typeahead.js?v=2.4.3:288:50)
    at Function.each (http://127.0.0.1:8080/js/jquery-3.6.1.min.js?v=2.4.3:2:3003)
    at S.fn.init.each (http://127.0.0.1:8080/js/jquery-3.6.1.min.js?v=2.4.3:2:1481)
    at $.fn.typeahead (http://127.0.0.1:8080/js/bootstrap/bootstrap-typeahead.js?v=2.4.3:284:17)
    at e.init_typeahead (http://127.0.0.1:8080/cluster-min.js?v=2.4.3:23:1079)
    at e.init_typeahead (http://127.0.0.1:8080/cluster-min.js?v=2.4.3:23:66)
    at e.start_router (http://127.0.0.1:8080/cluster-min.js?v=2.4.3:4:1990)
    at e.start_router (http://127.0.0.1:8080/cluster-min.js?v=2.4.3:4:171)
    at HTMLDocument.<anonymous> (http://127.0.0.1:8080/cluster-min.js?v=2.4.3:2:10490)
    at e (http://127.0.0.1:8080/js/jquery-3.6.1.min.js?v=2.4.3:2:30038)
    at t (http://127.0.0.1:8080/js/jquery-3.6.1.min.js?v=2.4.3:2:30340) undefined

Please Advise.

**Comment by srh** [comment]
It's broken.

jQuery was updated to address the complaint in #6937 , which is about a bug in jQuery which didn't affect the admin web UI, as far as I could tell (and I did take a close look).

If this is affecting your productivity, let us know...

**Comment by devtrix** [comment]
I am absolutely new to NoSQl and RethinkDB.  So far love it.  It would really help with our productivity indeed.  More than appreciation, our productivity will translate into a solid support for you guys.  I do appreciate all your efforts.

**Comment by v3ss0n** [comment]
> I am absolutely new to NoSQl and RethinkDB. So far love it. I tried rebuilding for Windows and its not going well. We use VS 2022 and all updated tools. I followed the instructions and still getting all kinds of errors. It would really help with our productivity indeed. More than appreciation, our productivity will translate into a solid support for you guys financially if given a chance. I do appreciate all your efforts.

Please open new issue if it's unrelated

**Comment by devtrix** [comment]
@v3ss0n My apologies for your confusion.  I was responding to @srh's question, it is affecting my productivity.  Further, I was elaborating as to the fact that I had trouble rebuilding for windows so that I could fix the jQuery error myself.  However, the tooling part you guys have is a bit outdated and I am sleeplessly working to release a major software called ControlDesk  myself so I have no time.  I am planning on using rethinkdb on this site and will gladly promote it as well.
So basically if this could be fixed on your side, great.  It will definitely help.
So not a new issue, just explaining what has happened.  Thanks!

**Comment by srh** [comment]
Thank you for giving fixing it a try.  Probably regenerating the web_assets.cc is something you'd want Linux for (see https://github.com/rethinkdb/rethinkdb/blob/v2.4.x/CONTRIBUTING.md ), and then rebuilding on Windows is another task.

Right now, given that nobody says they're going to fix it, you should anticipate the issue not being fixed.

**Comment by srh** [comment]
Taking a look at this seems like an important TODO for any future release.

**Comment by devtrix** [comment]
@srh Looks like the choices for .net drivers are in a sorry state.  I just cloned and upgraded the bchavez driver which was way outdated.  I am going to port it to .Net 7 and create Blazor addons as well.  I know this is not the place for this info but just wanted to let you guys know.

**Comment by srh** [comment]
@sg5506844 That explanation helps.

I made a fix in the admin ui implemented in https://github.com/srh/rethinkdb/tree/srh/typeahead

And the web_assets.cc is regenerated in https://github.com/srh/rethinkdb/tree/srh/gen_typeahead

I found jquery-migrate not to work.  Maybe it just doesn't support `$.browser`.  So I patched the bootstrap-typeahead library to use `'keydown'` events only and not `'keypress'`.  (That compatibility code is like, a decade old.)

I've tested it on Mac Chrome, Safari, and Firefox, and Linux Chrome and Firefox.  Once I test it on Windows browsers, I'll make PR's.

**Comment by srh** [comment]
Or we might use #7111...

**Comment by sg5506844** [comment]
Yes, I have created that PR, it uses jquery-browser.

**Comment by srh** [comment]
Closing as this is fixed with #7111 and #7113.

**Comment by besworks** [comment]
@srh Any chance you could find some time to pack this up into a release? This bug is definitely a productivity killer and makes it very difficult for new users to learn how to use the console.

**Comment by srh** [comment]
There is some chance.  It would be pretty lame not to.
